packages/compiler/src/core/checker.ts

```diff
@@ -1039,16 +1107,17 @@ export function createChecker(program: Program): Checker {
             })
           );
         }
-
-        if (sym.flags & SymbolFlags.LateBound) {
-          compilerAssert(sym.type, "Expected late bound symbol to have type");
-          return sym.type;
+        if (sym.flags & SymbolFlags.TemplateParameter) {
+          baseType = checkTemplateParameterDeclaration(
+            decl as TemplateParameterDeclarationNode,
+            mapper
+          );
         } else if (symbolLinks.declaredType) {
           baseType = symbolLinks.declaredType;
         } else if (sym.flags & SymbolFlags.Member) {
           baseType = checkMemberSym(sym, mapper);
         } else {
-          baseType = checkDeclaredType(sym, decl, mapper);
+          baseType = getTypeForNode(decl, mapper);
```

Comment:
is this change expected?

Reply:
All the checker changes were from when I thought I could fix everything without refactoring how symbols work.

Reply:
Ah make sense